verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -248,7 +251,43 @@ std::vector<verible::lsp::Location> SymbolTableHandler::FindDefinitionLocation(
   const verible::TextStructureView &text = parsedbuffer->parser().Data();
 
   const verible::TokenInfo cursor_token = text.FindTokenAt(cursor);
-  auto symbol = cursor_token.text();
+  return cursor_token.text();
+}
+
+std::optional<verible::lsp::Location>
+SymbolTableHandler::GetLocationFromSymbolTableNode(
+    absl::string_view symbol, const VerilogSourceFile *file_origin) {
+  // TODO (glatosinski) add iterating over multiple definitions
+  verible::lsp::Location location;
+  if (!file_origin && curr_project_) {
+    file_origin = curr_project_->LookupFileOrigin(symbol);
+  }
+  if (!file_origin) {
+    LOG(ERROR) << "Origin file not available";
+    return {};
+  }
+  location.uri = PathToLSPUri(file_origin->ResolvedPath());
+  auto *textstructure = file_origin->GetTextStructure();
+  if (!textstructure) {
+    LOG(ERROR) << "Origin file's text structure is not parsed";
+    return {};
+  }
+  verible::LineColumnRange symbollocation =
+      textstructure->GetRangeForText(symbol);
+  location.range.start = {.line = symbollocation.start.line,
+                          .character = symbollocation.start.column};
+  location.range.end = {.line = symbollocation.end.line,
+                        .character = symbollocation.end.column};
+  return location;
+}
+
+std::vector<verible::lsp::Location> SymbolTableHandler::FindDefinitionLocation(
+    const verible::lsp::DefinitionParams &params,
+    const verilog::BufferTrackerContainer &parsed_buffers) {
+  Prepare();
+  const absl::string_view filepath = LSPUriToPath(params.textDocument.uri);
+  std::string relativepath = curr_project_->GetRelativePathToSource(filepath);
+  auto symbol = GetTokenAtTextDocumentPosition(params, parsed_buffers);
```

Comment:
Try to reduce the use of `auto`, it makes it really hard after a while to read the code as the human is not as good as the computer to see what `auto` is.

Reply:
I fixed this in the `symbol-table-handler.cc` where possible (except for for-each loops)